datalad/support/gitrepo.py

```diff
@@ -999,7 +999,7 @@ def _get_prefixed_commit_msg(msg):
         return DATALAD_PREFIX if not msg else "%s %s" % (DATALAD_PREFIX, msg)
 
     def commit(self, msg=None, options=None, _datalad_msg=False, careless=True,
-               files=None):
+               files=None, date=None):
```

Comment:
Are you adding this argument just for testing?  If so, why not use the environment variables `GIT_AUTHOR_DATE` and `GIT_COMMITTER_DATE`?

Reply:
well, why should I use environment variables if command line has a dedicated option?

Reply:
Because you're adding an option that you otherwise have no use case for.  Also, this option only allows you to change the author, whereas but with the variables you could test both cases.

Reply:
could you fix it up to your liking? ;)  I was just trying to mimic cmdline, which ppl would be more familiar.
But indeed, having said that, we might want to keep in mind (not to be a part of this PR, but changes here should not preclude and only facilitate, thus env vars might be better) https://github.com/datalad/datalad/issues/1718 which relates and yet TODO.  interested to help with that one as well @kyleam -- it is needed for reproin ATM

Reply:
> could you fix it up to your liking? ;) I was just trying to mimic cmdline, which ppl would be more familiar.

It's fine as is, of course.  I was just curious.